core/templates/pages/blog-dashboard-page/blog-dashboard-tile/blog-dashboard-tile.component.spec.ts

```diff
@@ -97,4 +126,104 @@ describe('Blog Dashboard Tile Component', () => {
       expect(component.getDateStringInWords(DATE))
         .toBe('February 2, 2018');
     });
+
+  it('should navigate to blog post editor interface on clicking edit', () => {
+    spyOn(blogDashboardPageService, 'navigateToEditorTabWithId');
+    component.blogPostSummary = BlogPostSummary.createFromBackendDict(
+      sampleBlogPostSummary);
+
+    component.editBlogPost();
+
+    expect(blogDashboardPageService.navigateToEditorTabWithId)
+      .toHaveBeenCalledWith('sampleId');
+  });
+
+  it('should successfully place call to delete blog post model', fakeAsync(
+    () => {
+      spyOn(ngbModal, 'open').and.returnValue({
+        result: Promise.resolve()
+      } as NgbModalRef);
+      spyOn(blogDashboardPageService, 'deleteBlogPost');
+      component.blogPostSummary = BlogPostSummary.createFromBackendDict(
+        sampleBlogPostSummary);
+
+      component.deleteBlogPost();
+      tick();
+
+      expect(blogDashboardPageService.deleteBlogPost).toHaveBeenCalled();
+    }));
+
+  it('should cancel delete blog post model', fakeAsync(() => {
+    spyOn(ngbModal, 'open').and.returnValue({
+      result: Promise.reject()
+    } as NgbModalRef);
+    spyOn(blogDashboardPageService, 'deleteBlogPost');
+
+    component.deleteBlogPost();
+    tick();
+
+    expect(blogDashboardPageService.deleteBlogPost)
+      .not.toHaveBeenCalled();
+  }));
+
+  it('should unpublish blog post data successfully.', fakeAsync(() => {
+    spyOn(ngbModal, 'open').and.returnValue({
+      result: Promise.resolve()
+    } as NgbModalRef);
+
+    component.blogPostSummary = BlogPostSummary.createFromBackendDict(
+      sampleBlogPostSummary);
+    spyOn(blogPostEditorBackendApiService, 'updateBlogPostDataAsync')
+      .and.returnValue(Promise.resolve({blogPostDict: sampleBlogPostData}));
+    spyOn(component.unpublisedBlogPost, 'emit');
+
+    component.unpublishBlogPost();
+    tick();
```

Comment:
Ditto

Reply:
Same as above. We will resolve a promise from ngbModel.